VideoLocker/src/main/java/org/edx/mobile/module/prefs/UserPrefs.java

```diff
@@ -51,20 +52,21 @@ public boolean isDownloadOverWifiOnly() {
      */
     public File getDownloadFolder() {
         ProfileModel profile = getProfile();
-        
-        File android = new File(Environment.getExternalStorageDirectory(), "Android");
-        File downloadsDir = new File(android, "data");
-        File packDir = new File(downloadsDir, context.getPackageName());
-        File edxDir = new File(packDir, profile.username);
-        edxDir.mkdirs();
+        File videosDir = new File(context.getExternalFilesDir(null).getParent(), AppConstants.Folders.VIDEOS);
```

Comment:
Option 1 might be the simplest but that won't solve the issue. Plus, this change might seem out-of-scope but since I have to run migrations and do some renamings, so, I am actually stuck with fixing this.

I do have a solution in mind, which is to standardize the usage of app's external storage access using the FileUtil class, in which I'll add a function which will return the app's external storage directory. Something like this:

````java
    @NonNull
    public static File getAppExternalDir(@NonNull Context context) {
        File externalFilesDir = context.getExternalFilesDir(null);
        if (externalFilesDir != null) {
            return externalFilesDir.getParentFile();
        } else {
            File android = new File(Environment.getExternalStorageDirectory(), "Android");
            File downloadsDir = new File(android, "data");
            File packageDir = new File(downloadsDir, context.getPackageName());
            packageDir.mkdirs();
            return packageDir;
        }
    }
````